src/core/oauth2-authorize.js

```diff
@@ -66,6 +68,22 @@ export default function authorize ( { auth, authActions, errActions, configs, au
     query.push("realm=" + encodeURIComponent(authConfigs.realm))
   }
 
+  if (isCode && authConfigs.usePkceWithAuthorizationCodeGrant) {
+      // as per the PKCE spec, the Code Verifier must have a minumum length of 43 characters
+      // and random() produces a value that has a length of 32 characters
+      // source: https://tools.ietf.org/html/rfc7636#section-4.1
+      const codeVerifier = random() + random() + random()
```

Comment:
**Question**: is there a reason we're relying on a UUID library for this?

From § 4.1:

> It is RECOMMENDED that the output of
>  a suitable random number generator be used to create a 32-octet
>  sequence.  The octet sequence is then base64url-encoded to produce a
>  43-octet URL safe string to use as the code verifier.

Seems to me we could more easily do something like:

```js
// adapted from https://auth0.com/docs/flows/guides/auth-code-pkce/includes/create-code-verifier
function generateCodeVerifier() {
  return crypto.randomBytes(32)
    .toString('base64')
    .replace(/\+/g, '-')
    .replace(/\//g, '_')
    .replace(/=/g, '');
}
```

(Our Webpack config lets us use Node `crypto`, AFAIK)

Reply:
I like what you suggested better. I added a function at the end of the file and called it here.